### PR TITLE
feat: Support vLLM as a model backend

### LINKED ISSUE #639: [Feature Request] Efficiency improvement solution :VLLM
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

Study on efficiency improvement solutions and integrate necessary solutions

### Solution

VLLM: A high-throughput and memory-efficient inference and serving engine for LLMs

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zechengz**
## Description

Support vLLM as a model backend (by using [OpenAI Compatible Server](https://docs.vllm.ai/en/latest/serving/openai_compatible_server.html))

Fixes #639

## Motivation and Context

It is very useful for many projects and it's a little bit urgent

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/configs/vllm_config.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    n: int = 1
+    stream: bool = False
+    stop: str | Sequence[str] | NotGiven = NOT_GIVEN
+    max_tokens: int | NotGiven = NOT_GIVEN
+    presence_penalty: float = 0.0
+    response_format: dict | NotGiven = NOT_GIVEN
+    frequency_penalty: float = 0.0
+    logit_bias: dict = field(default_factory=dict)
+    user: str = ""
+
```
Comment (Appointat) [inline camel/configs/vllm_config.py:101]: Great, I have seen `tool_choice` and `tools` are not supported yet by vllm. We can keep an eye on it. When vllm is updated, we can submit a new PR.

Comment (Appointat) [reply camel/configs/vllm_config.py:101]: All looks good to me.

**Review by wangruohui — APPROVED** [review]
LGTM

**Review by Appointat — APPROVED** [review]
LGTM

**Pull request merged**